openedx/features/philu_utils/backend_storage.py

```diff
@@ -99,3 +101,17 @@ def get_alternative_name(self, file_root, file_ext):
     def exists(self, name):
         """Do not allow base class to update name in endless loop"""
         return False
+
+
+class ScormXblockS3Storage(CustomS3Storage):
+    def exists(self, name):
+        return super(ScormXblockS3Storage,self)._exists(name)
+
+    def open(self, name, mode='rb'):
+        client = self._s3_client()
+        bucket = client.get_bucket(self.aws_bucket_name)  # get bucket
+        key = bucket.get_key(name)  # get key (the file in s3)
+        fp = tempfile.TemporaryFile()
+        key.get_file(fp)
+        fp.seek(0)
```

Comment:
This will probably create a temp file on the server, but I am not sure about the intended behavior.

Also, it would be better to add unit tests, to test exact behavior

Reply:
"""Create and return a temporary file.
        Arguments:
        'prefix', 'suffix', 'dir' -- as for mkstemp.
        'mode' -- the mode argument to os.fdopen (default "w+b").
        'bufsize' -- the buffer size argument to os.fdopen (default -1).
        The file is created as mkstemp() would do it.

        Returns an object with a file-like interface.  The file has no
        name, and will cease to exist when it is closed.
        """

Reply:
@qasimgulzar bahi I meant to say was, what is the purpose of `open` function. Can we add doc string for it for calrification